src/hotspot/share/code/aotCodeCache.hpp

```diff
@@ -102,12 +101,12 @@ class AOTCodeEntry {
   uint   _size;        // Entry size
   uint   _name_offset; // Method's or intrinsic name
   uint   _name_size;
-  uint   _code_offset; // Start of code in cache
-  uint   _code_size;   // Total size of all code sections
-  uint   _reloc_offset;// Relocations
-  uint   _reloc_size;  // Max size of relocations per code section
   uint   _num_inlined_bytecodes;
 
+  uint   _blob_offset; // Start of archived blob in the cache
+  uint   _code_offset; // Start of code for an entry of type "Stub"
```

Comment:
Why you need 2 different fields for offset?

Reply:
I did this only for the clarity - using `_blob_offset` is for storing Stubs doesn't sound correct, so I added `_code_offset` for the Stubs. If it is confusing, I can keep `_blob_offset` and remove the `_code_offset` field.

Reply:
Or we can compromise: `_codeblob_offset, _codeblob_size`
Stubs are also using CodeBlob. I don't get why you treat stubs differently.

Reply:
StubGenerator Stubs used CodeBlob, but in AOT CodeCache we don't store their CodeBlobs as we do for adapters/runtime blobs. We instead only copy the assembly code for each stub.